lib/Adapter/TolerantParser/Fixer/IndentationFixer.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace Phpactor\CodeBuilder\Adapter\TolerantParser\Fixer;
+
+use Microsoft\PhpParser\Node;
+use Microsoft\PhpParser\Node\ClassMembersNode;
+use Microsoft\PhpParser\Node\DelimitedList;
+use Microsoft\PhpParser\Node\Statement\CompoundStatementNode;
+use Microsoft\PhpParser\Parser;
+use Phpactor\CodeBuilder\Domain\TextEdit;
+use Phpactor\CodeBuilder\Domain\StyleProposer;
+use Phpactor\CodeBuilder\Domain\TextEdits;
+use Phpactor\CodeBuilder\Util\TextFormat;
+use Phpactor\CodeBuilder\Util\TextUtil;
+
+class IndentationFixer implements StyleProposer
+{
+    /**
+     * @var Parser
+     */
+    private $parser;
+
+    /**
+     * @var string
+     */
+    private $indent;
+
+    public function __construct(Parser $parser, string $indent = '    ')
+    {
+        $this->parser = $parser;
+        $this->indent = $indent;
+    }
+
+    public function propose(string $text): TextEdits
+    {
+        $node = $this->parser->parseSourceFile($text);
+        $edits = $this->indentations($node, 0);
+
+        return TextEdits::fromTextEdits($edits);
+    }
+
+    private function indentations(Node $node, int $level): array
+    {
+        $edits = [];
+
+        $edits = $this->indent(
+            $edits,
+            $node,
+            $level
+        );
```

Comment:
Seems like you gave the responsibility the the `indent()` method to add the edit to the list when it should only be to create it.

Reply:
It can add zero or one -- so this avoids a conditional to check for a null. It's not amazing, but it's private.